### ISSUE TITLE
Usability of `select_sequential_consistency` setting.

### ISSUE THREAD

**Issue by alexey-milovidov** [ClickHouse/ClickHouse#5326]
When `select_sequential_consistency` is enabled, the user can get the following response:

```
Replica doesn't have part 20190515_20190515_4074_4074_0 which was successfully written to quorum of other replicas. Send query to another replica or disable 'select_sequential_consistency' setting.
```

That forces the client to implement retry logic on its own.
But it will be better to implement:
- HTTP redirect to in-sync replica for the HTTP interface;
- proxying to in-sync replica for native interface.

But the following considerations exist:

1. Redirections can lead to another redirections if a supposed in-sync replica became out of sync just in a moment.

2. If a query contains multiple tables, there can be no server with replicas for all tables in-sync at once.

3. For distributed queries, it's better to implement the choice of in-sync replica at the connection stage.

**Comment by sundy-li** [comment]
We got this issue,  this could assign to me.

**Comment by alexey-milovidov** [comment]
But it's already fixed in version 20.10.
Just enable `insert_quorum_parallel`.
